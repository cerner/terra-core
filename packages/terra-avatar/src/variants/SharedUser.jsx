import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from '../common/Avatar.module.scss';
import {
  AVATAR_VARIANTS, setColor,
} from '../common/AvatarUtils';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Specifies the alternative text for the image.
   */
  alt: PropTypes.string.isRequired,
  /**
   * Sets the background color. Defaults to `auto`. Accepted color variants are theme specific.
   * One of: `'auto'`, `'neutral'`, `'one'`, `'two'`, `'three'`, `'four'`, `'five'`, `'six'`, `'seven'`, `'eight'`, `'nine'`, `'ten'`.
   */
  color: PropTypes.oneOf(['auto', 'neutral', 'one', 'two', 'three', 'four',
    'five', 'six', 'seven', 'eight', 'nine', 'ten']),
  /**
   * Value used for the hash function when color is set to `auto`. If not provided, hash function utilizes alt.
   */
  hashValue: PropTypes.string,
  /**
   * Whether to hide avatar from the accessibility tree.
   */
  isAriaHidden: PropTypes.bool,
  /**
   * Overrides the default size.
   */
  size: PropTypes.string,
};

const defaultProps = {
  color: 'auto',
  hashValue: undefined,
  isAriaHidden: false,
  size: undefined,
};

const SharedUser = ({
  alt,
  color,
  hashValue,
  isAriaHidden,
  size,
  ...customProps
}) => {
  const colorVariant = setColor(alt, color, hashValue);
  const attributes = { ...customProps };
  const customStyles = size ? ({ fontSize: size, ...attributes.style }) : attributes.style;
  const multiUserClassNames = cx([
    'avatar',
    `${colorVariant}`,
    attributes.className,
  ]);

  const multiUserIconClassNames = cx(['icon', AVATAR_VARIANTS.SHARED_USER]);
  const multiUserContent = <span className={multiUserIconClassNames} role="img" aria-label={alt} alt={alt} aria-hidden={isAriaHidden} />;

  /* eslint-disable react/forbid-dom-props */
  return (
    <div {...attributes} className={multiUserClassNames} style={customStyles}>
      {multiUserContent}
    </div>
  );
  /* eslint-enable react/forbid-dom-props */
};

SharedUser.propTypes = propTypes;
SharedUser.defaultProps = defaultProps;

export default SharedUser;
