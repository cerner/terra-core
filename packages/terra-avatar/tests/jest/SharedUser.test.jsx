import React from 'react';
import SharedUser from '../../src/variants/SharedUser';

describe('SharedUser', () => {
  it('should render a default shared-user avatar', () => {
    const avatar = <SharedUser alt="shared-user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user avatar with provided size', () => {
    const avatar = <SharedUser alt="shared-user" size="5em" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user avatar with aria hidden', () => {
    const avatar = <SharedUser alt="shared-user" isAriaHidden />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user avatar with color variant one', () => {
    const avatar = <SharedUser alt="shared-user" color="one" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user with color variant neutral', () => {
    const avatar = <SharedUser alt="shared-user" color="neutral" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user with an automated color variant, based on a hashValue', () => {
    const avatar = <SharedUser alt="shared-user" hashValue="alternative hash" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a shared-user avatar with custom props', () => {
    const avatar = <SharedUser alt="shared-user" id="custom props shared user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });
});
