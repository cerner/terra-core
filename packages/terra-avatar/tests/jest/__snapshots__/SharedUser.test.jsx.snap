// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedUser should render a default shared-user avatar 1`] = `
<div
  class="avatar four"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user avatar with aria hidden 1`] = `
<div
  class="avatar four"
>
  <span
    alt="shared-user"
    aria-hidden="true"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user avatar with color variant one 1`] = `
<div
  class="avatar one"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user avatar with custom props 1`] = `
<div
  class="avatar four"
  id="custom props shared user"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user avatar with provided size 1`] = `
<div
  class="avatar four"
  style="font-size:5em"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user with an automated color variant, based on a hashValue 1`] = `
<div
  class="avatar two"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;

exports[`SharedUser should render a shared-user with color variant neutral 1`] = `
<div
  class="avatar neutral"
>
  <span
    alt="shared-user"
    aria-hidden="false"
    aria-label="shared-user"
    class="icon shared-user"
    role="img"
  />
</div>
`;
